### DISCUSSION TITLE
DSMR reading cannot be inserted into database

### DISCUSSION THREAD

**Discussion by eraaijmakers** [dsmrreader/dsmr-reader#1486] [Questions & answers / How-to's] [answered]
DSMR reader keeps trying to insert a reading into the database where it NULLs some fields where the database doesn't allow. This failure to insert keeps repeating itself and does not allow other readings to be inserted into the databse.

Is there some way to purge this one reading?

dsmr_backend.log
`[2021-02-25 16:32:12,039] ERROR    (IntegrityError) dsmr_consumption.services.run errored: null value in column "delivered_1" violates not-null constraint
DETAIL:  Failing row contains (901572, 2021-02-24 13:53:00+01, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null).

[2021-02-25 16:32:42,490] ERROR    (IntegrityError) dsmr_consumption.services.run errored: null value in column "delivered_1" violates not-null constraint
DETAIL:  Failing row contains (901573, 2021-02-24 13:53:00+01, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null).`

```

DSMR-READER
    App / Python / Database                                                       v4.12 / v3.9.1 / postgresql
    BE sleep / DL sleep / Retention / Override                                       1.0s / 5.0s / -h / False
    Latest telegram version read / Parser settings                                               "None" / "3"

DATA
    Telegrams total (est.)                                                                            5384941
    Consumption records electricity / gas (est.)                                               877310 / 14743

UNRESOLVED ISSUES
    Too many unprocessed readings: 9884                                                    1 day, 2 hours ago
    Day statistics are lagging behind                                                     1 day, 16 hours ago

POSTGRESQL SIZE OF LARGEST TABLES (> 500 MB)
    public.dsmr_datalogger_dsmrreading                                                                 993 MB

```

**Comment by dennissiemensma** [comment]
See #1288 and its references for more details and a solution

**Comment by eraaijmakers** [comment, marked as answer]
Removing the constraints and deleting the faulty records worked, there seemed to be only 1 faulty record. Constraints are put back in place and postgres:12.3-alpine is now used

Thanks!
